### ISSUE TITLE
Discussion: how do we submit transactions when running captive core?

### ISSUE THREAD

**Issue by abuiles** [stellar/go#2727]
Following this discussion https://stellarfoundation.slack.com/archives/C02B04RMK/p1592861365403100 

We need to define a strategy to submit transactions from front-end instances. Once #651 is done, front-end instances won't need access to Core's DB so they can communicate with a lighter version of stellar core (watcher).

Possible solutions:

- Have a single watcher per cluster to accept txs from front-end nodes.
- Run captive-cores on each front-end instance.
  - cons:  it will require a lot of RAM and will require using larger instances.
- Create a message bus between ingesting and front-end instances and have ingesting instances submit txs
  - cons: increased coupling between ingesting/front-end instances, we can't have read only replicas for front-end instances.

**Comment by abuiles** [comment]
I'm inclined towards a watches per cluster which is close to what we have today, this can be light instances of stellar-core since they won't need to keep tx meta.

**Comment by ire-and-curses** [comment]
Great points. An important scalability goal is to remove the 1:1 mapping between Horizon and Core. Read only replicas and horizontal scaling of public-facing Horizon nodes are a big win. It certainly seems wasteful to run multiple cores just so we can have a convenient submission endpoint.

We should be careful to distinguish between two deployment modes: cluster and stand-alone. For someone running a cluster, having a watcher core node for submissions doesn't seem like a big ask. For someone running a single Horizon instance, would it be possible to submit to the captive core? This seems like a nice usability balance if it can be done.

**Comment by bartekn** [comment]
I also think we should probably stick to a single watcher in case of a cluster with frontend/backend instances. In case of a single instance deployments I think captive instance could be used to tx submission.

In the long term I think it would be a good idea to at least consider implementing tx broadcast part of Stellar-Core's Overlay protocol. This way we completely remove a need for Stellar-Core to submit transactions. However, it requires that Stellar-Core avoids breaking changes to Overlay protocol.

**Comment by abuiles** [comment]
After discussing it here and offline  - we've reached the consensus that the best plan forward is to do what we described in the comments above:

1. Keep a  single watcher for a cluster with frontend/backend instances. Is up to the ops team how do they want to run it and  define a fail over architecture.
2. For single instance deployments, captive core can be used to submit transactions- when you run stellar-core in captive mode, it exposes the same http services as running on non-captive mode.
